lib/rubocop/cop/ezcater/graph_ql_type_file_order.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Ezcater
+      class GraphQlTypeFileOrder < Cop
+        ORDERED_DSL_METHODS = %w(
+          extend
+          include
+          description
+          pundit_policy_class
+          pundit_role
+          argument
+          field
+          user_errors_field
+        ).freeze
+
+        LOAD_METHOD_POSITION = ORDERED_DSL_METHODS.length
+        RESOLVE_METHOD_POSITION = ORDERED_DSL_METHODS.length + 1
+
+        METHOD_LOOKUP_LIST = ORDERED_DSL_METHODS.map { |method| ":#{method}" }.join(" | ")
+
+        def on_class(node)
+          gql_offenses = []
+          is_gql_type = false
+
+          type_dsl_calls(node).each_cons(2) do |previous, current|
+            previous = MatchedNode.new(previous)
+            current = MatchedNode.new(current)
+
+            if previous.dsl_method? || current.dsl_method?
+              is_gql_type = true
+            end
+
+            next unless previous > current
+
+            msg = "#{previous.name} should be positioned after #{current.name}"
```

Comment:
If I'm reading this right, it seems a little like a bubble sort. What happens if methods are misplaced beyond their nearest neighbor? e.g.:

```
argument
field
pundit_policy_class
pundit_role
description
```

Would a developer have to run several lint/fix cycles to get the code into the appropriate order? If so, perhaps it would be helpful to provide a reference to the Eng. Portal doc to help educate them?

Reply:
> If I'm reading this right, it seems a little like a bubble sort. 

You are, it is. This became even more apparent when I started trying to get this to autocorrect.

> What happens if methods are misplaced beyond their nearest neighbor? 

They would have to run this multiple times. It's not ideal at all, hence the reason I was poking at the autocorrect.

I'm totally behind adding a link to the output. It would be great if all cops provided that kind of explanation!